util/fork_func.c

```diff
@@ -0,0 +1,78 @@
+#include <sys/wait.h>
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include "warnp.h"
+
+#include "fork_func.h"
+
+/**
+ * fork_func(func, cookie):
+ * Fork and run ${func} in a new process, with ${cookie} as the sole argument.
+ */
+pid_t
+fork_func(int (* func)(void *), void * cookie)
+{
+	pid_t pid;
+
+	/* Fork */
+	switch (pid = fork()) {
+	case -1:
+		warnp("fork");
+		goto err0;
+	case 0:
+		/* In child process: Run the provided function, then exit. */
+		_exit((*func)(cookie));
```

Comment:
We generally omit the `*` when calling via a function pointer.  (The C standard says it's optional.)

Reply:
fixed